app/commands/git/sync_concept_exercise.rb

```diff
@@ -17,13 +17,29 @@ def call
         git_sha: head_git_exercise.commit.oid,
         synced_to_git_sha: head_git_exercise.commit.oid,
         taught_concepts: find_concepts(exercise_config[:concepts]),
-        prerequisites: find_concepts(exercise_config[:prerequisites])
+        prerequisites: find_concepts(exercise_config[:prerequisites]),
+        authors: authors
       )
+
+      update_reputation!
     end
 
     private
     attr_reader :exercise
 
+    def update_reputation!
+      exercise.authorships.each do |authorship|
+        User::ReputationAcquisition.find_or_create_by!(
+          user: authorship.author,
+          reason_object: authorship,
+          reason: "exercise_authorship",
+          category: "exercise_authorship"
+        ) do |ra|
+          ra.amount = 10
```

Comment:
We probably need to find a good location to define these reputation amounts. Would making them constants in `User::ReputationAcquisition` make sense as a location?

Reply:
Yeah. I think we want a constant hash of `{reason: amount}` and then in the `before_create` we can set `amount`. So the external API (ie, what you're calling here) only has to set the correct `reason`. We can also then check that the `reason` is valid based on whether it's a key in that hash.